### PR TITLE
Wallet RPC developer guide - Full refresh

### PR DESCRIPTION
**Pull request by el00ruobuob**
As i did for Daemon RPC dev guide, and as i did promise this for a long time, i started a full refresh of wallet RPC dev guide.

Here i will:

* add any missing call
* update existing call with:
  * new params
  * new examples
  * alias section
* reorder calls based on the wallet_rpc_server.h call order

I open this PR as a Work in Progress to track my progress:

* [x] get_balance
* [x] get_address
* [x] get_address_index
* [x] create_address
* [x] label_address
* [x] get_accounts
* [x] create_account
* [x] label_account
* [x] get_account_tags
* [x] tag_accounts
* [x] untag_accounts
* [x] set_account_tag_description
* [x] get_height
* [x] transfer
* [x] transfer_split
* [x] sign_transfer
* [x] submit_transfer
* [x] sweep_dust
* [x] sweep_all
* [x] sweep_single
* [x] relay_tx
* [x] store
* [x] get_payments
* [x] get_bulk_payments
* [x] incoming_transfers
* [x] query_key
* [x] make_integrated_address
* [x] split_integrated_address
* [x] stop_wallet
* [x] rescan_blockchain
* [x] set_tx_notes
* [x] get_tx_notes
* [x] set_attribute
* [x] get_attribute
* [x] get_tx_key
* [x] check_tx_key
* [x] get_tx_proof
* [x] check_tx_proof
* [x] get_spend_proof
* [x] check_spend_proof
* [x] get_reserve_proof
* [x] check_reserve_proof
* [x] get_transfers
* [x] get_transfer_by_txid
* [x] sign
* [x] verify
* [x] export_outputs
* [x] import_outputs
* [x] export_key_images
* [x] import_key_images
* [x] make_uri
* [x] parse_uri
* [x] get_address_book
* [x] add_address_book
* [x] refresh
* [x] delete_address_book
* [x] rescan_spent
* [x] start_mining
* [x] stop_mining
* [x] get_languages
* [x] create_wallet
* [x] open_wallet
* [x] change_wallet_password
* [x] is_multisig
* [x] prepare_multisig
* [x] make_multisig
* [x] export_multisig_info
* [x] import_multisig_info
* [x] finalize_multisig
* [x] sign_multisig
* [x] submit_multisig
* [x] get_version

Any suggestion here is welcome.

### CODE REVIEW

**Comment by artyomsol** [comment]
please consider to add `change_wallet_password` method [PR#4269](https://github.com/monero-project/monero/pull/4269)
```json
{"jsonrpc":"2.0","id":"0","method":"change_wallet_password","params":{"old_password":"theCurrentSecretPassPhrase","new_password":"theNewSecretPassPhrase"}}
```

**Comment by el00ruobuob** [comment]
@artyomsol i can easily consider it if:
- it gets merged;
- you build it for me on windows 64bits.

Then i can update my "based version", add the method, test it, and so on.

**Comment by el00ruobuob** [comment]
This is ready for review.
Call for developers @jtgrassie @glv2 @artyomsol @stoffu @naughtyfox @moneromooo-monero 

As it will take time to be reviewed, and if the few PR on RPC gets merged by then, i could have a final pass of update before it gets merged.

File: _i18n/en/resources/developer-guides/wallet-rpc.md [hunk 1]
```diff
@@ -142,1001 +181,1941 @@ Inputs:
...
+### **get_tx_proof**
+
+Get transaction signature to prove it.
+
+Alias: *None*.
+
+Inputs:
+
+* *txid* - string; transaction id.
+* *address* - string; destination publi
```
Comment (sneurlax) [inline _i18n/en/resources/developer-guides/wallet-rpc.md]: Is this no longer a comma-separated list of account indices?

Comment (el00ruobuob) [reply _i18n/en/resources/developer-guides/wallet-rpc.md]: I'm confused. Are you referring to get_tx_proof or get_reserve_proof ? I might have made a mistake anyway.

**Comment by sneurlax** [comment]
There's also a new `refresh` method as of https://github.com/monero-project/monero/pull/4242

File: _i18n/en/resources/developer-guides/wallet-rpc.md (hunk 1 above)
Comment (sneurlax) [reply _i18n/en/resources/developer-guides/wallet-rpc.md]: Sorry, those lines referenced above have nothing to do with this comment... I was referring to get_reserve_proof.  I was cross-referencing these notes with my own.  account_index isn't a comma-separated list, it's a number: https://github.com/monero-project/monero/blob/master/src/wallet/wallet_rpc_server.cpp#L2006

**Comment by el00ruobuob** [comment]
@sneurlax I count it in open PR needing a merge, and which i could update after that. I doubt this refresh will be reviewed fast, so i think if the PR is merged before 0.13 i could easily include it in the refresh.
The thing is, I do not want to add every upcoming thing until it gets merge so I can state on which master code the refresh is based.

**Comment by sneurlax** [comment]
Isn't it merged?  No worries, though, I just meant it as a reminder.

**Comment by el00ruobuob** [comment]
Oh yes it is. Just after my code freeze. And Luigi is merging things atm. So I might have a new pass really soon.

**Comment by jtgrassie** [comment]
There's also https://github.com/monero-project/monero/pull/4007 which maybe merged.

**Comment by el00ruobuob** [comment]
Noted.

File: _i18n/en/resources/developer-guides/wallet-rpc.md (hunk 1 above)
Comment (el00ruobuob) [reply _i18n/en/resources/developer-guides/wallet-rpc.md]: Ok, good.

**Comment by artyomsol** [comment]
@el00ruobuob `change_wallet_password` method monero-project/monero#4269 have been merged

**Comment by el00ruobuob** [comment]
Yes, I saw @luigi1111 merging a bunch of things. Also a param removed from transfer. Will give it an update tomorrow I think. But I won't be able to build on windows easily, so I might forge the examples.

**Comment by el00ruobuob** [comment]
Doing a new pass:
[ ] https://github.com/monero-project/monero/pull/4242
[x] https://github.com/monero-project/monero/pull/4263
[ ] https://github.com/monero-project/monero/pull/4269

**Comment by el00ruobuob** [comment]
@artyomsol @jtgrassie @sneurlax (and others) all the recent method are now documented.
Please let me know if you find anything wrong in this update.

**Comment by jtgrassie** [comment]
For `transfer` and `transfer_split` I think `account_index` and `subaddr_indices` are optional parameters. I.e. without specifying, defaults to primary account.

**Comment by el00ruobuob** [comment]
Corrected @jtgrassie

**Comment by el00ruobuob** [comment]
You are correct @sneurlax i corrected that. Thanks.

**Comment by sneurlax** [comment]
`getbalance` and `get_balance` are aliases of the same function. One is *actually* the alias, I suppose, but you can use either.

File: _i18n/ar/resources/developer-guides/wallet-rpc.md [hunk 2]
```diff
@@ -33,34 +33,62 @@ curl \
...
 Note: "@atomic-units" refer to the smallest fraction of 1 XMR according to the monerod implementation. **1 XMR = 1e12 @atomic-units.**
 
+This list has been updated on a frozen code on 2018-08-20 after merged commit b780cf4db1f9dfc49e7f16afc47892a5b40fe68a (Wallet RPC version 1.2), and at block height 1643308.
+
 ### Index of JSON RPC Methods:
 
-* [getbalance](#getbalance)
-* [getaddress](#getaddress)
-* [getheight](#getheight)
+* [get_balance](#get_balance)
```
[the remark this answers is not in the document]
Comment (sneurlax) [reply _i18n/ar/resources/developer-guides/wallet-rpc.md]: `getbalance` and `get_balance` are aliases of the same function. One is *actually* the alias, I suppose, but you can use either.

[the remark this answers is not in the document]
Comment (el00ruobuob) [reply _i18n/ar/resources/developer-guides/wallet-rpc.md]: I took the "underscored" in the command definition as being the regular as most of the calls uses underscores. But it doesn't change anything at all.

[the remark this answers is not in the document]
Comment (woodser) [reply _i18n/ar/resources/developer-guides/wallet-rpc.md]: Oh nice!  Definitely prefer to have the underscores then to be consistent.  Thanks.

[the remark this answers is not in the document]
Comment (el00ruobuob) [reply _i18n/ar/resources/developer-guides/wallet-rpc.md]: That was my thought too.

**Comment by woodser** [comment]
Most issues I had noted have been resolved by this PR.  I don't see anything immediately wrong.

Is there a reference document for the new methods or is this the first it's being pieced together?  Also how would one get a working implementation of these next gen methods?  Thanks.

**Comment by el00ruobuob** [comment]
I have based my work on current master code.
The new methods introduced since 0.12.3 will be available in 0.13.0.
You can compile current master if you want to use those.

**Comment by glv2** [comment]
The ```close_wallet``` RPC has been added in the master branch (https://github.com/monero-project/monero/commit/55b9acfce58eb808aa2814fc11c09ae17a51c7a3).

**Comment by el00ruobuob** [comment]
@glv2, i try to add it really soon.
BTW, the repo has bee moved on repo.getmonero.org, but this PR has some issues finding the branch over here.
I will open a new Merge Request soon, referring to the old one.

**Pull request closed without merging**